mypy/checkexpr.py

```diff
@@ -893,10 +894,61 @@ def analyze_ordinary_member_access(self, e: MemberExpr,
             return self.analyze_ref_expr(e)
         else:
             # This is a reference to a non-module attribute.
-            return analyze_member_access(e.name, self.accept(e.expr), e,
+            base_type = self.accept(e.expr)
+            member_type = analyze_member_access(e.name, base_type, e,
                                          is_lvalue, False, False,
                                          self.named_type, self.not_ready_callback, self.msg,
                                          chk=self.chk)
+            if is_lvalue:
+                return member_type
+            else:
+                return self.analyze_descriptor_access(base_type, member_type, e)
+
+    def analyze_descriptor_access(self, instance_type: Type, descriptor_type: Type,
+                                  context: Context) -> Type:
+        """
+        Type check descriptor access.
+
+        Return the inferred rvalue_type and whether to infer anything about the lvalue type
+        """
+        if not isinstance(descriptor_type, Instance):
+            return descriptor_type
+
+        if not descriptor_type.type.has_readable_member('__get__'):
+            return descriptor_type
+
+        dunder_get = descriptor_type.type.get_method('__get__')
+
+        if dunder_get is None:
+            self.msg.fail("{}.__get__ is not callable".format(descriptor_type), context)
+            return AnyType()
+
+        function = function_type(dunder_get, self.named_type('builtins.function'))
+        bound_method = bind_self(function, descriptor_type)
+        typ = map_instance_to_supertype(descriptor_type, dunder_get.info)
+        dunder_get_type = expand_type_by_instance(bound_method, typ)
+
+        if isinstance(instance_type, FunctionLike) and instance_type.is_type_obj():
```

Comment:
I think I covered both places that I was checking `is_type_obj` with a check for `TypeType` as well (and added tests to that effect)